File: crypto/shared/src/main/scala/io/iohk/atala/crypto/MnemonicException.scala
package io.iohk.atala.crypto

sealed class MnemonicException(message: String, cause: Throwable) extends Exception(message, cause)

case class MnemonicLengthException(message: String, cause: Throwable) extends MnemonicException(message, cause)

case class MnemonicWordException(message: String, cause: Throwable) extends MnemonicException(message, cause)

case class MnemonicChecksumException(message: String, cause: Throwable) extends MnemonicException(message, cause)

File: prism-kotlin-sdk/crypto/src/commonMain/kotlin/io/iohk/atala/prism/kotlin/crypto/derivation/MnemonicException.kt
package io.iohk.atala.prism.kotlin.crypto.derivation

import kotlin.js.JsExport

@JsExport
sealed class MnemonicException(message: String?, cause: Throwable?) :
    Exception(message, cause)

@JsExport
class MnemonicLengthException(message: String?, cause: Throwable? = null) :
    MnemonicException(message, cause)

@JsExport
class MnemonicWordException(message: String?, cause: Throwable? = null) :
    MnemonicException(message, cause)

@JsExport
class MnemonicChecksumException(message: String?, cause: Throwable? = null) :
    MnemonicException(message, cause)
